### PR TITLE
add error description and ID to image api 

### PR DESCRIPTION
**Pull request by mgold1234**
# Description
add error description and ID to image api 

FIXES: https://issues.redhat.com/browse/THEEDGE-3347
https://issues.redhat.com/browse/THEEDGE-3381
https://issues.redhat.com/browse/THEEDGE-3401

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update
- [ ] Tests update
- [ ] Refactor

### CODE REVIEW

File: pkg/models/images_api.go [hunk 1]
```diff
@@ -7,18 +7,18 @@ type CommitAPI struct {
...
 type CustomPackagesAPI struct {
-	Name string `json:"name" example:"cat"`
+	Name string `json:"name" example:"cat"` // Name of custom packages
 } // @name CustomPackages
 
 // InstallerAPI ...
 type InstallerAPI struct {
-	SSHKey   string `json:"sshkey" example:"ssh-rsa lksjdkjkjsdlkjjds"`
-	Username string `json:"username" example:"myuser"`
+	SSHKey   string `json:"sshkey" example:"ssh-rsa lksjdkjkjsdlkjjds"` // SSh key of installer
```
Comment (loadtheaccumulator) [inline pkg/models/images_api.go:15]: typo /Ssh/SSH/

File: pkg/models/images_api.go [hunk 2]
```diff
@@ -7,18 +7,18 @@ type CommitAPI struct {
...
+	Username string `json:"username" example:"myuser"`                  // Username of installer
```
Comment (loadtheaccumulator) [inline pkg/models/images_api.go:16]: Suggesting...
// Username of administrative user

File: pkg/models/images_api.go [hunk 3]
```diff
@@ -30,20 +30,19 @@ type ThirdPartyReposAPI struct {
 
 // CreateImageAPI is the /images POST endpoint struct for openapi.json auto-gen
 type CreateImageAPI struct {
-	Commit         CommitAPI           `json:"commit"`
-	CustomPackages []CustomPackagesAPI `json:"customPackages"`                                 // An optional list of custom repositories
+	Commit         CommitAPI           `json:"commit"`                                         // commit of image
+	CustomPackages []CustomPackagesAPI `json:"customPackages" example:"Name:'customPackage'"`  // An optional list of custom repositories
```
Comment (loadtheaccumulator) [inline pkg/models/images_api.go:34]: This is probably a good time to correct the original cut/paste error :-)
// An optional list of custom packages

This is also of type []CustomePackagesAPI so the example will need to be a list.

File: pkg/models/images_api.go [hunk 4]
```diff
@@ -30,20 +30,19 @@ type ThirdPartyReposAPI struct {
...
 	Description    string              `json:"description" example:"This is an example image"` // A short description of the image
 	Distribution   string              `json:"distribution" example:"rhel-92"`                 // The RHEL for Edge OS version
 	// Available image types:
 	// * rhel-edge-installer - Installer ISO
 	// * rhel-edge-commit - Commit only
 	ImageType              string               `json:"imageType" example:"rhel-edge-installer"` // The image builder assigned image type
 	Installer              InstallerAPI         `json:"installer"`
-	Name                   string               `json:"name"  example:"my-edge-image"`
-	Packages               []PackagesAPI        `json:"packages"`
```
Comment (loadtheaccumulator) [inline pkg/models/images_api.go:43]: CreateImage() loops through the Packages list, so the Packages line is needed.
https://github.com/loadtheaccumulator/edge-api/blob/e8f8d8b366a46e206ff7308e9b0338fb18b74baa/pkg/services/images.go#L315

**Review by loadtheaccumulator — CHANGES_REQUESTED** [review]
I just have a handful of suggestions/changes. Otherwise, looks good.

File: pkg/models/images_api.go (hunk 2 above)
Comment (mgold1234) [reply pkg/models/images_api.go]: changed, thank you

**Review by loadtheaccumulator — APPROVED** [review]
lgtm

**Review by acosferreira — APPROVED** [review]
lgtm

**Pull request merged**